mlflow/store/artifact/http_artifact_repo.py

```diff
@@ -1,32 +1,30 @@
 import os
-import requests
 import posixpath
 
 from mlflow.entities import FileInfo
 from mlflow.store.artifact.artifact_repo import ArtifactRepository, verify_artifact_path
 from mlflow.utils.file_utils import relative_path_to_artifact_path
-from mlflow.utils.rest_utils import augmented_raise_for_status
+from mlflow.utils.rest_utils import MlflowHostCreds, augmented_raise_for_status, http_request
 
 
 class HttpArtifactRepository(ArtifactRepository):
     """Stores artifacts in a remote artifact storage using HTTP requests"""
 
     def __init__(self, artifact_uri):
         super().__init__(artifact_uri)
-        self._session = requests.Session()
+        self.host_creds = self._get_host_creds(artifact_uri)
```

Comment:
Can we make this a private attribute? 

```suggestion
        self._host_creds = self._get_host_creds(artifact_uri)
```

Reply:
Done:)